cosmos/x/evm/plugins/state/keys.go

```diff
@@ -27,6 +27,13 @@ import (
 
 // NOTE: we use copy to build keys for max performance: https://github.com/golang/go/issues/55905
 
+func BalanceKeyFor(address common.Address) []byte {
+	bz := make([]byte, 1+common.AddressLength)
+	copy(bz, []byte{types.BalanceKeyPrefix})
+	copy(bz[1:], address[:])
+	return bz
```

Comment:
```suggestion
	return append([]byte{types.BalanceKeyPrefix}, address[:]...)
```
the `[:]` also does a copy